tests/legacy-cli/e2e/utils/fs.ts

```diff
@@ -1,195 +1,145 @@
+import { stripIndents } from 'common-tags';
 
-export function writeFile(fileName: string, content: string, options?: any) {
-  return new Promise<void>((resolve, reject) => {
-    fs.writeFile(fileName, content, options, (err: any) => {
-      if (err) {
-        reject(err);
-      } else {
-        resolve();
-      }
-    });
-  });
+export function readFile(fileName: string): Promise<string> {
+  return fs.readFile(fileName, 'utf-8');
 }
 
-
-export function deleteFile(path: string) {
-  return new Promise<void>((resolve, reject) => {
-    fs.unlink(path, (err) => {
-      if (err) {
-        reject(err);
-      } else {
-        resolve();
-      }
-    });
-  });
+export function writeFile(fileName: string, content: string, options?: any): Promise<void> {
+  return fs.writeFile(fileName, content, options);
 }
 
-
-export function rimraf(path: string) {
-  return new Promise<void>((resolve, reject) => {
-    fs.remove(path, (err?: any) => {
-      if (err) {
-        reject(err);
-      } else {
-        resolve();
-      }
-    });
-  });
+export function deleteFile(path: string): Promise<void> {
+  return fs.unlink(path);
 }
 
-
-export function moveFile(from: string, to: string) {
-  return new Promise<void>((resolve, reject) => {
-    fs.rename(from, to, (err) => {
-      if (err) {
-        reject(err);
-      } else {
-        resolve();
-      }
-    });
-  });
+export function rimraf(path: string): Promise<void> {
+  return fs.rmdir(path, { recursive: true, maxRetries: 3 });
 }
 
-
-export function symlinkFile(from: string, to: string, type?: string) {
-  return new Promise<void>((resolve, reject) => {
-    fs.symlink(from, to, type, (err) => {
-      if (err) {
-        reject(err);
-      } else {
-        resolve();
-      }
-    });
-  });
+export function moveFile(from: string, to: string): Promise<void> {
+  return fs.rename(from, to);
 }
 
-export function createDir(path: string) {
-  return _recursiveMkDir(path);
+export function symlinkFile(from: string, to: string, type?: string): Promise<void> {
+  return fs.symlink(from, to, type);
 }
 
-
-function _recursiveMkDir(path: string): Promise<void> {
-  if (fs.existsSync(path)) {
-    return Promise.resolve();
-  } else {
-    return _recursiveMkDir(dirname(path))
-      .then(() => fs.mkdirSync(path));
-  }
+export function createDir(path: string): Promise<void> {
+  return fs.mkdir(path, { recursive: true });
 }
 
-export function copyFile(from: string, to: string) {
-  return _recursiveMkDir(dirname(to))
-    .then(() => new Promise((resolve, reject) => {
-      const rd = fs.createReadStream(from);
-      rd.on('error', (err: Error) => reject(err));
+export async function copyFile(from: string, to: string): Promise<void> {
+  await createDir(dirname(to));
 
-      const wr = fs.createWriteStream(to);
-      wr.on('error', (err: Error) => reject(err));
-      wr.on('close', () => resolve());
+  return new Promise<void>((resolve, reject) => {
```

Comment:
This can probably be reduced to:
`return fs.copyFile(from, to, constants.COPYFILE_FICLONE);`